File: java/src/main/java/robocup/control/GameService.java
package robocup.control;

import java.util.HashMap;
import java.util.HashSet;
import java.util.Map;
import java.util.Set;

import javax.enterprise.context.ApplicationScoped;

@ApplicationScoped
public class GameService {
    private Map<Integer, Game> games = new HashMap<>();

    public Set<Integer> listGameIds() {
        return new HashSet<Integer>(games.keySet());
    }

    public void setGame(Game game) {
        games.put(game.getGameId(), game);
    }

    public Game getGame(int gameId) {
        return games.get(gameId);
    }
}

File: cpp/game_service.cc
#include "game_service.h"

#include <iostream>

#include "game.h"

std::list<int> GameService::listGameIds() {
  std::list<int> ids;
  for (const auto &[key, value] : games_) {
    ids.push_back(key);
  }
  return ids;
}

void GameService::setGame(std::unique_ptr<Game> game) {
  games_[game->game_id] = std::move(game);
}

Game *GameService::getGame(int game_id) noexcept {
  auto idx = games_.find(game_id);
  if (idx != games_.end()) {
    return idx->second.get();
  } else {
    return nullptr;
  }
}